### PR TITLE
Merge src/Native/build.sh into top level build.sh

### LINKED ISSUE #3155: Building native shims from top level folder
Currently there is no top level way to build the native shims. One has to call src/Native/build.sh. Since the shims are needed for CoreFX to work correctly on Unix systems, it would be helpful to provide some intuitive way of invoking the build from the top level.

I see two obvious ways to archive this:

1. Create a top level build-native.sh which starts the build in src/Native
2. Have the top level build.sh invoking building the shims in addition to the manages assemblies.

I am in favour of option 2. How are general ideas?

cc @nguerrera @sokket @stephentoub

(By mistake I opened this issue on coreclr as ~~dotnet/coreclr#1527~~ at first)

### PR DESCRIPTION
**Pull request by janhenke**
If no argument is passed, defaults to build both the managed and the
native code. The src/Native/build.sh is kept to not break CI, it simply
calls the top level build.sh to call the native part. Closes #3155 

I tested on Linux.

### CODE REVIEW

**Comment by janhenke** [comment]
@mmitche The native PR tests fail due to the way how src/Native/build.sh is called. I left the file there, which just calls the top level build.sh now. Shall I fix this problem or can we agree on updating the PR check configuration to simply use the top level build.sh? In theory we also do not need separate jobs for the managed and native parts any more.

**Comment by jasonwilliams200OK** [comment]
:+1: 
Perhaps we can remove `src/Native/build.sh` and send a PR for @mmitche fixing https://github.com/dotnet/dotnet-ci/search?utf8=%E2%9C%93&q=src%2FNative%2Fbuild.sh&type=Code.

**Comment by jasonwilliams200OK** [comment]
@dotnet-bot, test this please.

**Comment by janhenke** [comment]
Okay I have updated the PR and dropped the other build.sh completely. I will add a PR to the dotnet/dotnet-ci repository to update the job configurations accordingly.

**Comment by janhenke** [comment]
Okay this PR has to be merged together with dotnet/dotnet-ci#59, then everything should work again and we have proper "one builds it all" support.

File: build.sh [hunk 1]
```diff
@@ -1,96 +1,353 @@
...
+    OpenBSD)
+        __BuildOS=OpenBSD
+        ;;
+
+    NetBSD)
+        __BuildOS=NetBSD
+        ;;
+
+    *)
+        echo "Unsupported OS $OSName detected, configuring as if for Linux"
```
Comment (nguerrera) [inline build.sh]: I know this is how it was, but we should make this an error and remove OpenBSD and NetBSD until there's an actual effort to port to those.

**Comment by nguerrera** [comment]
I would do these three changes sequentially:

1. This change + src/Native/build.sh stub that works when invoked from any directory (*)
2. dotnet/dotnet-ci#59
3. Delete src/Native/build.sh stub

That way, there's no race to get this and dotnet/dotnet-ci#59 in together.

(*) I believe all that was needed was the same __scriptDir approach that was already there instead of using ../.. from the current directory.

**Comment by mmitche** [comment]
@jasonmalinowski @janhenke Unfortunately a PR sent to dotnet-ci won't directly affect the CI.  It's mostly currently used a backup location.  There is stuff in the works to shift jobs to being generated by the Job DSL plugin which will enable this kind of workflow, but we're not there yet.  For now I'd go with Nick's approach

File: build.sh (hunk 1 above)
Comment (janhenke) [inline build.sh]: I would not make it an hard error. But I am okay with removing OpenBSD and NetBSD for now.

**Comment by janhenke** [comment]
I have updated the PR. The src/Native/build.sh is back as a stub calling the top level build.sh. This time this should work from the top level as well. So we can forward with the three step plan as mentioned by Nick above.

**Comment by jasonwilliams200OK** [comment]
:+1: for @nguerrera's approach. Thanks @janhenke for the prompt update.
@mmitche, is it so that in future we will have access to Job DSL setting in repository and we can include CI changes within the related PR? That would be nice. This is exactly how other GitHub repositories do which make use of TravisCI+AppVeyor stack. :+1:

**Comment by mmitche** [comment]
@jasonwilliams200OK Yeah that's the plan.  I'm thinking that most of the configuration will still go into dotnet-ci, so that a lot of common configuration parameters can be pulled out.

File: build.sh [hunk 2]
```diff
@@ -1,96 +1,345 @@
 #!/usr/bin/env bash
 
+usage()
+{
+    echo "Usage: $0 [managed] [native] [BuildArch] [BuildType] [clean] [verbose] [clangx.y]"
+    echo "manged - optional argument to build the managed code"
+    echo "native - optional argument to build the native code"
+    echo "The following arguments affect native builds only:"
```
Comment (nguerrera) [inline build.sh]: Just a thought for future, but Clean, Verbose, and BuildType could (and probably should) be made to work with managed too.

**Comment by nguerrera** [comment]
LGTM. Can someone else take a look as well? 

cc @ellismg

**Comment by ellismg** [comment]
LGTM.  Thanks @janhenke.  @mmitche Are you okay with this being merged now?

File: build.sh (hunk 2 above)
Comment (janhenke) [inline build.sh]: Correct, but I tried to keep the delta small with this PR. We can always extend the functionality from this point on.

**Comment by stephentoub** [comment]
@mmitche, can this be merged?

**Comment by mmitche** [comment]
Yes.

**Pull request merged**